### PR TITLE
Fix num_token_padding support for static per-tensor scaled_fp8_quant

### PR DESCRIPTION
**Pull request by mgoin**
## Purpose

FIX https://github.com/vllm-project/vllm/issues/20192

After https://github.com/vllm-project/vllm/pull/20076 fixed the assert to always check for per-tensor scales in the static scaled_fp8_quant case, this test `tests/quantization/test_fp8.py::test_scaled_fp8_quant` started failing
https://buildkite.com/vllm/ci/builds/22749#0197ad88-59ca-41d0-9692-2bb3f5c6dca3

It seems to me that there is no reason why padding wouldn't be supported in this static per-tensor quant case

## Test Plan

Use the existing failing test at
https://github.com/vllm-project/vllm/blob/aafabaa0d5c87c283b366f81fdce55cf91ae980c/tests/quantization/test_fp8.py#L190-L196

## Test Result

```
tests/quantization/test_fp8.py::test_scaled_fp8_quant[dtype0] PASSED
tests/quantization/test_fp8.py::test_scaled_fp8_quant[dtype1] PASSED
```

### CODE REVIEW

**Review by tlrmchlsmth — APPROVED** [review]

**Comment by mgoin** [comment]
Validated on B200 and H100 (for marlin)

Command:
```
lm_eval --model vllm --model_args pretrained=nm-testing/Qwen3-30B-A3B-NVFP4,enforce_eager=True --trust_remote_code --tasks gsm8k --num_fewshot 5 --batch_size auto
```

B200:
```
vllm (pretrained=nm-testing/Qwen3-30B-A3B-NVFP4,enforce_eager=True,trust_remote_code=True), gen_kwargs: (None), limit: None, num_fewshot: 5, batch_size: auto
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8908|±  |0.0086|
|     |       |strict-match    |     5|exact_match|↑  |0.8878|±  |0.0087|
```

H100:
```
vllm (pretrained=nm-testing/Qwen3-30B-A3B-NVFP4,enforce_eager=True,trust_remote_code=True), gen_kwargs: (None), limit: None, num_fewshot: 5, batch_size: auto
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8855|±  |0.0088|
|     |       |strict-match    |     5|exact_match|↑  |0.8795|±  |0.0090|
```

File: vllm/_custom_ops.py [hunk 1]
```diff
@@ -1274,8 +1274,7 @@ def scaled_fp8_quant(
             scale = torch.zeros(1, device=input.device, dtype=torch.float32)
             torch.ops._C.dynamic_scaled_fp8_quant(output, input, scale)
     else:
-        # num_token_padding not implemented for this case
-        assert (scale.numel() == 1 and num_token_padding is None)
+        assert scale.numel() == 1
```
Comment (minosfuture) [inline vllm/_custom_ops.py:1277]: this assertion depends on `use_per_token_if_dynamic`. if `use_per_token_if_dynamic`, scale has a shape `(shape[0], 1)`. 

`tests/kernels/moe/test_cutlass_moe.py` on main is currently failing due to #20076 as well.
